lib/ansible/module_utils/aws/elbv2.py

```diff
@@ -0,0 +1,757 @@
+
+# Non-ansible imports
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+except ImportError:
+    pass
+import traceback
+import time
+from copy import deepcopy
+
+
+class ElasticLoadBalancerV2(object):
+
+    def __init__(self, connection, module):
+
+        self.connection = connection
+        self.module = module
+        self.changed = False
+        self.new_load_balancer = False
+        self.scheme = module.params.get("scheme")
+        self.name = module.params.get("name")
+        self.subnet_mappings = module.params.get("subnet_mappings")
+        self.subnets = module.params.get("subnets")
+        self.deletion_protection = module.params.get("deletion_protection")
+        self.wait = module.params.get("wait")
+
+        if module.params.get("tags") is not None:
+            self.tags = ansible_dict_to_boto3_tag_list(module.params.get("tags"))
+        else:
+            self.tags = None
+        self.purge_tags = module.params.get("purge_tags")
+
+        self.elb = get_elb(connection, module, self.name)
+        if self.elb is not None:
+            self.elb_attributes = self.get_elb_attributes()
+            self.elb['tags'] = self.get_elb_tags()
+        else:
+            self.elb_attributes = None
+
+    def wait_for_status(self, elb_arn):
+        """
+        Wait for load balancer to reach 'active' status
+
+        :param elb_arn: The load balancer ARN
+        :return:
+        """
+
+        try:
+            waiter = self.connection.get_waiter('load_balancer_available')
+            waiter.wait(LoadBalancerArns=[elb_arn])
+        except (BotoCoreError, ClientError) as e:
+            self.module.fail_json_aws(e)
+
+    def get_elb_attributes(self):
+        """
+        Get load balancer attributes
+
+        :return:
+        """
+
+        try:
+            attr_list = AWSRetry.jittered_backoff()(
+                self.connection.describe_load_balancer_attributes
+            )(LoadBalancerArn=self.elb['LoadBalancerArn'])['Attributes']
+
+            elb_attributes = boto3_tag_list_to_ansible_dict(attr_list)
+        except (BotoCoreError, ClientError) as e:
+            self.module.fail_json_aws(e)
+
+        # Replace '.' with '_' in attribute key names to make it more Ansibley
+        return dict((k.replace('.', '_'), v) for k, v in elb_attributes.items())
+
+    def update_elb_attributes(self):
+        """
+        Update the elb_attributes parameter
+        :return:
+        """
+        self.elb_attributes = self.get_elb_attributes()
+
+    def get_elb_tags(self):
+        """
+        Get load balancer tags
+
+        :return:
+        """
+
+        try:
+            return AWSRetry.jittered_backoff()(
+                self.connection.describe_tags
+            )(ResourceArns=[self.elb['LoadBalancerArn']])['TagDescriptions'][0]['Tags']
+        except (BotoCoreError, ClientError) as e:
+            self.module.fail_json_aws(e)
```

Comment:
Can we have helpful messages when failing? I tend to like `module.fail_json_aws(e, "Couldn't get tags")` but perhaps I'm being fussy and perhaps it's obvious from the stacktrace.

Reply:
I always find AWS messages good enough and a lot of the other modules i've written rely on that.  I haven't seen any noise in the issues to make me think that other users don't think the same.  Happy to accomodate though.